### PR TITLE
Script fixes to clarify and further automate the release process

### PR DESCRIPTION
**Pull request by jthaine**

### CODE REVIEW

File: release/README-development-process.html [hunk 1]
```diff
@@ -55,15 +55,29 @@ <h2 align="center">Checker Framework Development Process</h2>
...
+                   <ol>
+                     <li>
+                               Log into Buffalo<br/><br/>
+                               <code>ssh $USER@buffalo.cs.washington.edu<br/></code></br>
+                     </li>
+                     <li>
+                               In a user-specific temporary directory, clone the Checker Framework repository (it contains the release scripts).<br/>
+                               If the repository exists, update it instead.<br/><br/>
+                               <code>mkdir -p /scratch/$USER/jsr308-release<br/></code>
+                               <code>cd /scratch/$USER/jsr308-release<br/></code>
```
Comment (mernst) [inline release/README-development-process.html]: It's even easier to give the user a single set of commands to run, with no thinking required.  This works:

    mkdir -p /scratch/$USER/jsr308-release
    /scratch/$USER/jsr308-release
    git -C checker-framework pull || git clone https://github.com/typetools/checker-framework.git
    cd checker-framework/release

File: release/release_build.py [hunk 2]
```diff
@@ -245,44 +250,37 @@ def main(argv):
...
-    # be sure to carefully read all instructions on the command-line before typing yes. This is because
-    # the scripts do not ask you to say 'yes' after each step, so you may miss a step if you only read
-    # the paragraph asking you to say 'yes'.
+    delete_if_exists( RELEASE_BUILD_COMPLETED_FLAG_FILE )
 
     set_umask()
 
     projects_to_release = read_projects( argv, print_usage )
+
+    #Check for a --auto
```
Comment (mernst) [inline release/release_build.py]: I suggest inserting a space between the comment syntax "#" and the following text.

File: release/release_build.py [hunk 3]
```diff
@@ -326,80 +324,84 @@ def main(argv):
...
 
-    print("If any checkers have been added or removed, then verify that the lists")
-    print("of checkers in these manual sections are up to date:")
-    print(" * Introduction")
-    print(" * Run-time tests and type refinement")
+    print_step("Build Step 4: Copy entire live site to dev site (~22 minutes).") # AUTO
 
-    print("Verify that the manual PDF has no lines that are longer than the page width")
-    print("(it is acceptable for some lines to extend into the right margin).")
+    # TODO: Ask support to delete the dev_old directory as it has files I cannot delete or chmod, then remove "--exclude=dev_old" from the command below.
```
Comment (mernst) [inline release/release_build.py]: Can you go ahead and do this?

Comment (jthaine) [inline release/release_build.py]: Request submitted. I now have enough confidence to delete the old dev site after having tested the build script a few times.

**Pull request merged**